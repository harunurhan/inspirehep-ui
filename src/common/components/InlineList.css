.__InlineList__ {
  display: inline;
  list-style: none;
  padding: 0;
}

.__InlineList__ li {
  display: inline;
}

.__InlineList__ li:not(:last-child):after {
  content: "; ";
}
