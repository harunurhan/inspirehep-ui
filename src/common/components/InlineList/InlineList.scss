.__InlineList__ {
  ul {
    display: inline;
    list-style: none;
    padding: 0;
  }
  li {
    display: inline;
  }
  ul.separate-items {
    li:not(:last-child):after {
      content: '; ';
    }
  }
}
