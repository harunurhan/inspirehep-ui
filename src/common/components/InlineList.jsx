import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { List } from 'immutable';

import './InlineList.css';

class InlineList extends Component {
  render() {
    const {
      items, renderItem, label, suffix,
    } = this.props;
    return items && (
      <div >
        {label && <span>{label}: </span>}
        <ul className="__InlineList__">
          {items.map(item => (
            // TODO: use proper key
            <li key={item} style={{ display: 'inline' }}>{renderItem(item)}</li>
          ))}
        </ul>
        {suffix}
      </div>
    );
  }
}

InlineList.propTypes = {
  items: PropTypes.instanceOf(List),
  renderItem: PropTypes.func.isRequired,
  suffix: PropTypes.node,
  label: PropTypes.string,
};

InlineList.defaultProps = {
  label: null,
  items: null,
  suffix: null,
};

export default InlineList;
