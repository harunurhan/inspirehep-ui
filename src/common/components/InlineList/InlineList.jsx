import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { List } from 'immutable';

import './InlineList.scss';

class InlineList extends Component {
  render() {
    const {
      items, renderItem, label, suffix, extractKey,
    } = this.props;
    return items && (
      <div className="__InlineList__">
        {label && <span>{label}: </span>}
        <ul>
          {items.map(item => (
            <li key={extractKey(item)}>{renderItem(item)}</li>
          ))}
        </ul>
        {suffix}
      </div>
    );
  }
}

InlineList.propTypes = {
  items: PropTypes.instanceOf(List),
  renderItem: PropTypes.func.isRequired,
  extractKey: PropTypes.func,
  suffix: PropTypes.node,
  label: PropTypes.string,
};

InlineList.defaultProps = {
  label: null,
  items: null,
  suffix: null,
  extractKey: item => item,
};

export default InlineList;
