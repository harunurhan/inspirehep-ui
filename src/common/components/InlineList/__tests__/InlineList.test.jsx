import React from 'react';
import { shallow } from 'enzyme';
import { fromJS } from 'immutable';

import InlineList from '../InlineList';

describe('InlineList', () => {
  it('renders with minimal props set', () => {
    const items = fromJS(['foo', 'bar']);
    const wrapper = shallow((
      <InlineList
        items={items}
        renderItem={item => <span>{item}</span>}
      />
    ));
    expect(wrapper).toMatchSnapshot();
  });

  it('renders with all props set', () => {
    const items = fromJS([
      { id: 1, value: 'foo' },
      { id: 2, value: 'bar' },
    ]);
    const wrapper = shallow((
      <InlineList
        label="Test"
        suffix={<span>Suffix</span>}
        items={items}
        extractKey={item => item.get('id')}
        renderItem={item => <span>{item.get('value')}</span>}
      />
    ));
    expect(wrapper).toMatchSnapshot();
  });

  it('does not render if items are null', () => {
    const wrapper = shallow((
      <InlineList
        label="Test"
        suffix={<span>Suffix</span>}
        items={null}
        renderItem={item => <span>{item}</span>}
      />
    ));
    expect(wrapper).toMatchSnapshot();
  });
});
