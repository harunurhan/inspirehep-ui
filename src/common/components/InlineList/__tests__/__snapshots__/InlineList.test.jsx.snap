// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InlineList does not render if items are null 1`] = `""`;

exports[`InlineList renders with all props set 1`] = `
<div
  className="__InlineList__"
>
  <span>
    Test
    : 
  </span>
  <ul>
    <li
      key="1"
    >
      <span>
        foo
      </span>
    </li>
    <li
      key="2"
    >
      <span>
        bar
      </span>
    </li>
  </ul>
  <span>
    Suffix
  </span>
</div>
`;

exports[`InlineList renders with minimal props set 1`] = `
<div
  className="__InlineList__"
>
  <ul>
    <li
      key="foo"
    >
      <span>
        foo
      </span>
    </li>
    <li
      key="bar"
    >
      <span>
        bar
      </span>
    </li>
  </ul>
</div>
`;
